core/services/ocr2/plugins/mercury/integration_test.go

```diff
@@ -184,15 +184,17 @@ func integration_MercuryV1(t *testing.T) {
 	bootstrapNode := Node{App: appBootstrap, KeyBundle: bootstrapKb}
 	logObservers = append(logObservers, observedLogs)
 
-	// Commit blocks to finality depth to ensure LogPoller has finalized blocks to read from
-	ch, err := bootstrapNode.App.GetRelayers().LegacyEVMChains().Get(testutils.SimulatedChainID.String())
-	require.NoError(t, err)
-	finalityDepth := ch.Config().EVM().FinalityDepth()
-	for i := 0; i < int(finalityDepth); i++ {
-		backend.Commit()
-	}
-
-	fromBlock := int(finalityDepth) // cannot use zero, start from finality depth
+	// cannot use zero, start from finality depth
+	fromBlock := func() int {
```

Comment:
why doesn't this cause cascading changes to the call site of fromBlock.

iiuc it changes the type, so i would have expected a diff fromBlock -> fromBlock()

Reply:
Hmmm I'm not following. The only effect I intended here was to scope the error var so that it couldn't even be accessed from down below. `fromBlock` is still an `int`, and I just cast when returning.